### PR TITLE
Cleanup MD.Core and MD.Ide references.

### PR DESCRIPTION
**Pull request by KirillOsenkov**
Move most Roslyn references to MD.Ide.
Remove some unneeded references from MD.Core.
Sort references in MD.Core and MD.Ide.
Remove ProductVersion and SchemaVersion which are never useful.

### CODE REVIEW

**Comment by KirillOsenkov** [comment]
I've verified that the actual references that end up baked into MD.Core.dll and MD.Ide.dll are the same as before. If it builds and the CI build passes it should give us enough confidence that I didn't mess anything up. I tried to be careful when transplanting references and sorting.

**Comment by mkrueger** [comment]
It's ok to leave some references in core

File: main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj [hunk 1]
```diff
@@ -134,24 +112,41 @@
...
+    </Reference>
+    <Reference Include="Mono.Cecil.Rocks">
+      <HintPath>..\..\..\packages\Mono.Cecil.0.10.0-beta5\lib\net40\Mono.Cecil.Rocks.dll</HintPath>
+    </Reference>
+    <Reference Include="monodoc, Version=1.0.0.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756" />
+    <Reference Include="Mono.Posix" />
+    <Reference Include="mscorlib" />
+    <Reference Include="Newtonsoft.Json">
+      <HintPath>..\..\..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
+    </Reference>
```
Comment (mrward) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj]: We seem to have two different Newtonsoft.Json versions being referenced. Is this going to cause a problem? MonoDevelop.Core references 9.0.1 whilst MonoDevelop.Ide references 8.0.3. I think we should be using 8.0.3 everywhere.

Comment (KirillOsenkov) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Core.csproj]: This is a good point. I'll send a separate PR to unify all Newtonsoft.Json references to 8.0.3.

**Pull request merged**